src/client/datascience/jupyter/kernels/cellExecutionQueue.ts

```diff
@@ -141,6 +141,13 @@ export class CellExecutionQueue implements Disposable {
                 await this.cancel();
                 break;
             }
+            // If the kernel is dead, then no point trying the rest.
+            if (kernelConnection.status === 'dead' || kernelConnection.status === 'terminating') {
```

Comment:
Cancel all pending executions, no point even trying

Reply:
What could happen is, if kernel dies and we have more cells, things go into a weird start & execution never completes.